Userland/Libraries/LibWeb/WebSockets/WebSocket.cpp

```diff
@@ -55,18 +56,39 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<WebSocket>> WebSocket::construct_impl(JS::R
         return WebIDL::SyntaxError::create(realm, "Invalid protocol");
     if (!url_record.fragment().is_empty())
         return WebIDL::SyntaxError::create(realm, "Presence of URL fragment is invalid");
-    // 5. If `protocols` is a string, set `protocols` to a sequence consisting of just that string
-    // 6. If any of the values in `protocols` occur more than once or otherwise fail to match the requirements, throw SyntaxError
-    return MUST_OR_THROW_OOM(realm.heap().allocate<WebSocket>(realm, window, url_record));
+    Vector<DeprecatedString> protocols_sequence;
+    if (protocols.has_value()) {
+        // 5. If `protocols` is a string, set `protocols` to a sequence consisting of just that string
+        if (protocols.value().has<DeprecatedString>())
+            protocols_sequence.append(protocols.value().get<DeprecatedString>());
```

Comment:
> 5. If `protocols` is a string, set `protocols` to a sequence consisting of just that string
```suggestion
            protocols_sequence = { protocols.value().get<DeprecatedString>() };
```

I think it's clearer with the spec comment above if you directly test the vector to just contain that string. Seeing `.append()` it's not immediately obvious that's what's happening.